pulp_smash/tests/rpm/api_v2/test_upload_publish.py

```diff
@@ -149,6 +149,78 @@ def _get_pkglist(update):
     return out_pkglist
 
 
+class UploadDrpmTestCase(utils.BaseAPITestCase):
+    """Test whether one can upload and associate DRPMs."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Upload DRPM to a repo.
+
+        Steps:
+
+        1. Create a RPM repo.
+        2. Upload DRPM to repo.
+        3. Check if DRPM has been"uploaded.
+
+        https://github.com/PulpQE/pulp-smash/issues/336
+        """
+        super(UploadDrpmTestCase, cls).setUpClass()
+        cls.responses = {}
+
+        # init repo and client
+        client = api.Client(cls.cfg, api.json_handler)
+        repo = client.post(REPOSITORY_PATH, gen_repo())
+        cls.resources.add(repo['_href'])
+        client.response_handler = api.safe_handler
+
+        # Download DRPM
+        cls.drpm = utils.http_get(DRPM_URL)
+
+        # Start upload
+        cls.responses['malloc'] = client.post(CONTENT_UPLOAD_PATH)
+        cls.responses['upload'] = client.put(
+            urljoin(cls.responses['malloc'].json()['_href'], '0/'),
+            data=cls.drpm,
+        )
+        cls.responses['import'] = client.post(
+            urljoin(repo['_href'], 'actions/import_upload/'),
+            {
+                'unit_key': {},
+                'unit_type_id': 'drpm',
+                'upload_id': cls.responses['malloc'].json()['upload_id'],
+            },
+        )
+        # Remove upload task
+        cls.responses['free'] = client.delete(
+            cls.responses['malloc'].json()['_href'],
+        )
+
+        # Search for drpm to determine if is present
+        body = {'criteria': {}}
+        cls.responses['repo units'] = client.post(urljoin(repo['_href'], 'search/units/'), body)
+
+    def test_status_code(self):
```

Comment:
I think there's already a test case which verifies the status codes returned by each of these actions.

Please drop this test method entirely, and use the `pulp_smash.utils.upload_import_unit` function instead.